### PR TITLE
Remove GCS Interface from WorkflowDescriptor.  Closes #498

### LINKED ISSUE #498: Remove gcsInterface: Try[GoogleCloudStorage] from WorkflowDescriptor

### PR DESCRIPTION
**Pull request by scottfrazer**
- [x] Run Tyburn tests

### CODE REVIEW

**Comment by mcovarr** [comment]
@kshakir and I

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala [hunk 1]
```diff
@@ -83,6 +82,12 @@ case class WorkflowDescriptor(id: WorkflowId,
   to ensure expected behavior. The NioGcsPaths could easily retain pointers to their parent paths, even when relative.
    */
 
+  lazy val gcsInterface = ioManager.interfaces.collect({ case gcs: GoogleCloudStorage => gcs }) match {
+    case Seq(gcs: GoogleCloudStorage) => Success(gcs)
+    case s: Seq[GoogleCloudStorage] if s.size > 1 => Failure(new UnsupportedOperationException(s"More than one GCS interface found: $s"))
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: don't really need the `: Seq[GoogleCloudStorage]` here

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala [hunk 2]
```diff
@@ -83,6 +82,12 @@ case class WorkflowDescriptor(id: WorkflowId,
   to ensure expected behavior. The NioGcsPaths could easily retain pointers to their parent paths, even when relative.
    */
 
+  lazy val gcsInterface = ioManager.interfaces.collect({ case gcs: GoogleCloudStorage => gcs }) match {
+    case Seq(gcs: GoogleCloudStorage) => Success(gcs)
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: don't appear to need this `: GoogleCloudStorage`

**Comment by mcovarr** [comment]
:+1:

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala [hunk 3]
```diff
@@ -83,6 +82,12 @@ case class WorkflowDescriptor(id: WorkflowId,
   to ensure expected behavior. The NioGcsPaths could easily retain pointers to their parent paths, even when relative.
    */
 
+  lazy val gcsInterface = ioManager.interfaces.collect({ case gcs: GoogleCloudStorage => gcs }) match {
+    case Seq(gcs) => Success(gcs)
+    case s if s.size > 1 => Failure(new UnsupportedOperationException(s"More than one GCS interface found: $s"))
+    case  _ => Failure(new UnsupportedOperationException(s"No GCS interface found"))
```
Comment (kshakir) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: Two spaces in between `case  _`.

**Comment by kshakir** [comment]
One character to delete, then :+1:

**Pull request merged**